diff --git a/node_modules/@greenwood/plugin-postcss/src/types/index.d.ts b/node_modules/@greenwood/plugin-postcss/src/types/index.d.ts
index 9fb23c3..7d4a641 100644
--- a/node_modules/@greenwood/plugin-postcss/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-postcss/src/types/index.d.ts
@@ -1,10 +1,10 @@
-import type { Plugin } from "@greenwood/cli";
+import type { Plugin, ResourcePlugin } from "@greenwood/cli";
 
 type PostCssPluginOptions = {
   extendConfig?: boolean;
 };
 
-export type PostCssPlugin = (options?: PostCssPluginOptions) => Array<Plugin>;
+export type PostCssPlugin = (options?: PostCssPluginOptions) => ResourcePlugin;
 
 declare module "@greenwood/plugin-postcss" {
   export const greenwoodPluginPostCss: PostCssPlugin;
