diff --git a/node_modules/@greenwood/plugin-google-analytics/src/types/index.d.ts b/node_modules/@greenwood/plugin-google-analytics/src/types/index.d.ts
index 3eabd6e..8f37493 100644
--- a/node_modules/@greenwood/plugin-google-analytics/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-google-analytics/src/types/index.d.ts
@@ -1,6 +1,11 @@
-import type { Plugin } from "@greenwood/cli";
+import type { Plugin, ResourcePlugin } from "@greenwood/cli";
 
-export type GoogleAnalyticsPlugin = () => Array<Plugin>;
+export type GoogleAnalyticsPluginOptions = {
+  analyticsId: string;
+  anonymous?: boolean;
+}
+
+export type GoogleAnalyticsPlugin = (options: GoogleAnalyticsPluginOptions) => ResourcePlugin;
 
 declare module "@greenwood/plugin-google-analytics" {
   export const greenwoodPluginGoogleAnalytics: GoogleAnalyticsPlugin;
